native-sql-engine/core/src/main/scala/org/apache/spark/sql/execution/python/ColumnarArrowPythonRunner.scala

```diff
@@ -64,10 +64,11 @@ class ColumnarArrowPythonRunner(
       startTime: Long,
       env: SparkEnv,
       worker: Socket,
+      pid: Option[Int],
       releasedOrClosed: AtomicBoolean,
       context: TaskContext): Iterator[ColumnarBatch] = {
 
-    new ReaderIterator(stream, writerThread, startTime, env, worker, releasedOrClosed, context) {
```

Comment:
Keep these two versions of newReaderIterator for either implementing the interface of spark 3.1 or spark 3.2.